### PR TITLE
Error kind detection bug fix 

### LINKED ISSUE #1753: Gas Issue while Withdrawing from Supernet
# Gas Issue while Withdrawing from Supernet
## Description
Hi, I have followed the guide [here](https://github.com/integrations-Polygon/Supernets-Edge-Tutorials/blob/master/Supernets_v1.0.0_setup.md) in order to run a local Supernet. 

After running the 4 validators I use the following command to deposit a Rootchain ERC20 to the Supernet:
```
./polygon-edge bridge deposit-erc20 --amounts 200000000000000000 --json-rpc $MUMBAI_RPC --receivers 0xAE3ac03497aB6aFfF786bCFF95DD6e9095Dc5d55 --root-predicate <erc20PredicateAddress from genesis.json>--root-token 0xaADBC81Dec3D3f27a23b9fC8d9F1f031034079e6 --sender-key $PRIVATE_KEY
```
The following output is returned, suggesting the deposit was a success
```
Sender                  = 0xAE3ac03497aB6aFfF786bCFF95DD6e9095Dc5d55
Receivers               = 0xAE3ac03497aB6aFfF786bCFF95DD6e9095Dc5d55
Amounts                 = 200000000000000000
Inclusion Block Numbers = 38317126
Child Token Address     = 0x3EB10E0c47Cab33036199f5e20bc89C4B2Cca72B
```

Next I use the following command to withdraw:
```
./polygon-edge bridge withdraw-erc20 --amounts 100000000000000000 --child-predicate <erc20ChildMintablePredicateAddress from genesis.json> --child-token 0x3EB10E0c47Cab33036199f5e20bc89C4B2Cca72B --json-rpc http://127.0.0.1:9545 --receivers 0xAE3ac03497aB6aFfF786bCFF95DD6e9095Dc5d55 --sender-key $PRIVATE_KEY
```

However, I receive the following error on doing so
```
failed to send withdraw transaction (receiver: 0xAE3ac03497aB6aFfF786bCFF95DD6e9095Dc5d55, amount: 100000000000000000). error: {"code":-32603,"message":"not enough gas supplied for intrinsic gas costs"})
```

I have ensured that the sender address, validator addresses, and the 0x0 address on the Supernet are funded.  However, we still receive this error. Kindly let me know if I have used incorrect flags in the genesis command or there is something else that can be done here. I'd like to which address does not have enough gas as suggested in the error.

## Your environment
- Ubuntu 22 LTS.
- Latest Commit on dev branch, hash bbbe3ab.

### PR DESCRIPTION
**Pull request by begmaroman**
# Description

Addressing: https://github.com/0xPolygon/polygon-edge/issues/1753

Error `state.ErrNotEnoughIntrinsicGas` was wrapped into the high level error wrapper and was not properly detected using `errors.Is` function

# Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [x] I have assigned this PR to myself
- [x] I have added at least 1 reviewer
- [x] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [x] I have tested this code manually

### CODE REVIEW

File: command/bridge/deposit/erc20/deposit_erc20.go [hunk 1]
```diff
@@ -155,7 +155,7 @@ func runCommand(cmd *cobra.Command, _ []string) {
 
 	receipt, err := txRelayer.SendTransaction(approveTxn, depositorKey)
 	if err != nil {
-		outputter.SetError(fmt.Errorf("failed to send root erc 20 approve transaction"))
+		outputter.SetError(fmt.Errorf("failed to send root erc 20 approve transaction: %w", err))
```
Comment (Stefan-Ethernal) [inline command/bridge/deposit/erc20/deposit_erc20.go]: Heads up: this is addressed via #1768.

File: state/executor.go [hunk 2]
```diff
@@ -523,7 +523,7 @@ var (
 	ErrNotEnoughFundsForGas  = fmt.Errorf("not enough funds to cover gas costs")
 	ErrBlockLimitReached     = fmt.Errorf("gas limit reached in the pool")
 	ErrIntrinsicGasOverflow  = fmt.Errorf("overflow in intrinsic gas calculation")
-	ErrNotEnoughIntrinsicGas = fmt.Errorf("not enough gas supplied for intrinsic gas costs")
+	ErrNotEnoughIntrinsicGas = errors.New("not enough gas supplied for intrinsic gas costs")
```
Comment (Stefan-Ethernal) [inline state/executor.go:526]: Should we use `errors.New` format for other errors as well?

**Review by Stefan-Ethernal — APPROVED** [review]

File: state/executor.go (hunk 2 above)
Comment (begmaroman) [reply state/executor.go:526]: Makes sense, done

**Review by goran-ethernal — APPROVED** [review]

**Review by vcastellm — APPROVED** [review]
LGTM

**Pull request merged**